MAINT: Rename sample-peds, feature-peds, pprs and peds-simulation

cherman2 opened the pull request:
sample-peds -> pedf
 feature-peds -> prdf
pprs -> pprf
peds-simulation -> peds-permutation-test

cherman2 committed:
renaming all the things

cherman2 committed:
Merge branch 'dev' of https://github.com/qiime2/q2-fmt into finishing-touches

lizgehret commented:
ok so just making sure i understand - you're also removing the original peds method since you've fully replaced this functionality with pedf and prdf, correct?

cherman2 commented:
Yep! I don't have multiple 'peds' methods anymore so a pipeline that can run multiple peds methods felt weird so I decided to remove it in favor of pedf and prdf.

lizgehret commented on q2_fmt/_peds.py:
```diff
@@ -102,7 +70,7 @@ def heatmap(output_dir: str, data: pd.DataFrame,
     if "all possible recipients with feature" in data.columns:
         if drop_incomplete_subjects or drop_incomplete_timepoints:
-            warnings.warn('Feature PEDS was selected as the PEDS metric, which'
+            warnings.warn('PRDF was selected as the porportion metric, which'
```
```suggestion
            warnings.warn('PRDF was selected as the proportion metric, which'
```

lizgehret commented on q2_fmt/_peds.py:
```diff
@@ -216,49 +184,49 @@ def sample_peds(table: pd.DataFrame, metadata: qiime2.Metadata,
     with warnings.catch_warnings():
         warnings.simplefilter("ignore")
-        peds = median_numerator_series/median_denominator_series
+        pedf = median_numerator_series/median_denominator_series
```
The real meat and potatoes ➗

lizgehret commented on q2_fmt/_peds.py:
```diff
@@ -216,49 +184,49 @@ def sample_peds(table: pd.DataFrame, metadata: qiime2.Metadata,
-    peds_df['measure'].attrs.update({
-        'title': "Sample PEDS",
+    pedf_df['measure'].attrs.update({
+        'title': "Sample pedf",
```
```suggestion
        'title': "Sample PEDF",
```
Should this be in all caps, like PEDS was before?

lizgehret commented on q2_fmt/_peds.py:
```diff
@@ -316,38 +284,38 @@ def feature_peds(table: pd.DataFrame, metadata: qiime2.Metadata,
     })
-    peds_df['measure'].attrs.update({
+    prdf_df['measure'].attrs.update({
         'title': "Feature PEDS",
```
```suggestion
        'title': "Feature PRDF",
```
I'm assuming this title should get updated as well?

lizgehret commented on q2_fmt/_peds.py:
```diff
@@ -365,33 +333,33 @@ def _compute_peds(peds_df: pd.Series, peds_type: str, peds_time: int,
                                 reference_column=reference_column)
     maskedrecip = donormask & recip_df
-    if peds_type == "Sample" or peds_type == "PPRS":
+    if type == "Sample" or type == "PPRS":
```
```suggestion
    if type == "Sample" or type == "PPRF":
```

lizgehret commented on q2_fmt/_peds.py:
```diff
@@ -365,33 +333,33 @@ def _compute_peds(peds_df: pd.Series, peds_type: str, peds_time: int,
+                               sample_row[time_column]]
 
-    elif peds_type == "Feature":
+    elif type == "Feature":
```
Should this stay as-is? or type be changed to PRDF?

cherman2 replied:
I was keeping the internals the same because I figured it didn't matter, but I will change it

lizgehret commented on q2_fmt/_peds.py:
```diff
@@ -427,25 +395,25 @@ def sample_pprs(table: pd.DataFrame, metadata: qiime2.Metadata,
-                                time_column=time_column,
-                                subject_column=subject_column,
-                                reference_column=used_references.name)
+        pprf_df = _compute_proportion(df=pprf_df, type='PPRS',
```
```suggestion
        pprf_df = _compute_proportion(df=pprf_df, type='PPRF',
```

lizgehret commented on q2_fmt/_peds.py:
```diff
@@ -455,51 +423,51 @@ def sample_pprs(table: pd.DataFrame, metadata: qiime2.Metadata,
-    peds_df['measure'].attrs.update({
+    pprf_df['measure'].attrs.update({
         'title': 'PPRS',
         'description': 'Proportional Persistence of Recipient Strains'
```
```suggestion
        'description': 'Proportional Persistence of Recipient Features'
```

lizgehret commented on q2_fmt/_peds.py:
```diff
@@ -585,12 +552,12 @@ def peds_simulation(table: pd.DataFrame, metadata: qiime2.Metadata,
     # Denominator for PEDS Calc. (Number of unique features in the Donor)
```
```suggestion
    # Denominator for PEDF Calc. (Number of unique features in the Donor)
```

lizgehret commented on q2_fmt/plugin_setup.py:
```diff
@@ -233,48 +233,14 @@
             'per_subject_stats': StatsTable[Pairwise],
             'global_stats': StatsTable[Pairwise]},
-    input_descriptions={'data': 'PEDS or PPRS output to plot',
+    input_descriptions={'data': 'PEDF or PPRS output to plot',
```
```suggestion
    input_descriptions={'data': 'PEDF or PPRF output to plot',
```

lizgehret commented on q2_fmt/plugin_setup.py:
```diff
@@ -285,22 +251,22 @@
         'drop_incomplete_subjects': drop_incomplete_subjects},
     name=' Proportional Features Heatmap',
-    description='Plot heatmap for PEDS or PPRS value over time',
+    description='Plot heatmap for PEDF, PRDF or PPRS value over time',
```
```suggestion
    description='Plot heatmap for PEDF, PRDF or PPRF value over time',
```

lizgehret commented on q2_fmt/plugin_setup.py:
```diff
@@ -310,28 +276,30 @@
-         ' recipient sample',
+    name='Proportional Engraftment of Donor Features in each'
+         ' recipient sample. This is adapted from aggarwala et al. 2021'
+         ' stainer manuscript, which coined the term PEDS, PEDF uses the same'
```
```suggestion
         ' stainer manuscript, which coined the term PEDS. PEDF uses the same'
```

lizgehret commented on q2_fmt/plugin_setup.py:
```diff
@@ -342,28 +310,28 @@
+        'prdf_dists': pedf_dists
     },
-    name='Porportional Engraftment of Donor Strains per feature',
+    name='Porportion of Recipient with Donor Feature',
```
```suggestion
    name='Porportion of Recipients with Donor Feature',
```
Not sure if this should be pluralized? Ignore if not

lizgehret commented on q2_fmt/plugin_setup.py:
```diff
@@ -342,28 +310,28 @@
     },
     output_descriptions={
-        'peds_dists': peds_dists
+        'prdf_dists': pedf_dists
```
Should this be `prdf_dists` instead of `pedf_dists`?

cherman2 replied:
this is a hard question i guess, 
This dist are literally the same and ive always used the same "string" named peds_dist as the output description. I'll change it but I am not sure how yet

lizgehret commented on q2_fmt/plugin_setup.py:
```diff
@@ -374,21 +342,23 @@
-         ' recipient sample',
+    name='Proportional Persistence of Recipient Features in each'
+         ' recipient sample. This is adapted from aggarwala et al. 2021'
+         ' stainer manuscript, which coined the term PPRS, PPRF uses the same'
```
```suggestion
         ' stainer manuscript, which coined the term PPRS. PPRF uses the same'
```

lizgehret commented on q2_fmt/plugin_setup.py:
```diff
@@ -414,31 +384,31 @@
                             ' This will make it faster to run this method but'
                             ' the actual values may be slightly less'
                             ' comparable to the simulated values which will'
                             ' be undergo rarefaction `num_resamples` of times',
```
```suggestion
                            ' undergo rarefaction `num_resamples` of times',
```

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -791,12 +791,13 @@ def test_get_donor(self):
-                                table=table_df, metadata=metadata_df,
-                                time_column="group", reference_column="Ref",
-                                subject_column="subject")
+        peds_df = _compute_proportion(df=peds_df, type="Sample",
```
```suggestion
        pedf_df = _compute_proportion(df=pedf_df, type="Sample",
```
also wondering if you're keeping 'sample' as the type? i know the others were updated to the abbreviation, so not sure if you wanted to be consistent there (since this is just internal - not that users will see it)

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -791,12 +791,13 @@ def test_get_donor(self):
+                                      time_column="group",
+                                      reference_column="Ref",
+                                      subject_column="subject")
         peds_df = peds_df.set_index("id")
```
```suggestion
        pedf_df = pedf_df.set_index("id")
```

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -791,12 +791,13 @@ def test_get_donor(self):
+                                      reference_column="Ref",
+                                      subject_column="subject")
         peds_df = peds_df.set_index("id")
         donor = peds_df.at["sample1", "donor"]
```
```suggestion
        donor = pedf_df.at["sample1", "donor"]
```

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -822,12 +823,13 @@ def test_get_subject(self):
-                                table=table_df, metadata=metadata_df,
-                                time_column="group", reference_column="Ref",
-                                subject_column="subject")
+        peds_df = _compute_proportion(df=peds_df, type="Sample",
```
```suggestion
        pedf_df = _compute_proportion(df=pedf_df, type="Sample",
```
same comment as above re: sample vs PEDF as type

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -822,12 +823,13 @@ def test_get_subject(self):
+                                      time_column="group",
+                                      reference_column="Ref",
+                                      subject_column="subject")
         peds_df = peds_df.set_index("id")
```
```suggestion
        pedf_df = pedf_df.set_index("id")
```

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -822,12 +823,13 @@ def test_get_subject(self):
+                                      reference_column="Ref",
+                                      subject_column="subject")
         peds_df = peds_df.set_index("id")
         subject = peds_df.at["sample1", "subject"]
```
```suggestion
        subject = pedf_df.at["sample1", "subject"]
```

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -853,12 +855,13 @@ def test_timepoint(self):
-                                table=table_df, metadata=metadata_df,
-                                time_column="group", reference_column="Ref",
-                                subject_column="subject")
+        peds_df = _compute_proportion(df=peds_df, type="Sample",
```
```suggestion
        pedf_df = _compute_proportion(df=pedf_df, type="Sample",
```

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -853,12 +855,13 @@ def test_timepoint(self):
+                                      time_column="group",
+                                      reference_column="Ref",
+                                      subject_column="subject")
         peds_df = peds_df.set_index("id")
```
```suggestion
        pedf_df = pedf_df.set_index("id")
```

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -853,12 +855,13 @@ def test_timepoint(self):
+                                      reference_column="Ref",
+                                      subject_column="subject")
         peds_df = peds_df.set_index("id")
         tp = peds_df.at["sample3", "group"]
```
```suggestion
        tp = pedf_df.at["sample3", "group"]
```

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -950,10 +953,10 @@ def test_no_feature_overlap(self):
+                       time_column="group",
+                       reference_column="Ref",
+                       subject_column="subject")
         exp_peds_df = pd.DataFrame({
```
```suggestion
        exp_pedf_df = pd.DataFrame({
```

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -963,7 +966,7 @@ def test_no_feature_overlap(self):
             'measure': [0, 0.333333, 1, 1]
             })
-        pd.testing.assert_frame_equal(sample_peds_df, exp_peds_df)
+        pd.testing.assert_frame_equal(pedf_df, exp_peds_df)
```
```suggestion
        pd.testing.assert_frame_equal(pedf_df, exp_pedf_df)
```

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -1188,12 +1191,12 @@ def test_reference_series_not_in_table(self):
-                          table=table_df, metadata=metadata_df,
-                          time_column="group", reference_column="Ref",
-                          subject_column="subject")
+            _compute_proportion(df=peds_df, type="Sample",
```
```suggestion
            _compute_proportion(df=pedf_df, type="Sample",
```

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -1417,7 +1420,7 @@ def test_pprs(self):
             'measure': [0.0, 1.0, 0.0, 1.0]
             })
-        pd.testing.assert_frame_equal(sample_pprs_df, exp_pprs_df)
+        pd.testing.assert_frame_equal(pprf_df, exp_pprs_df)
```
```suggestion
        pd.testing.assert_frame_equal(pprf_df, exp_pprf_df)
```

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -1402,11 +1405,11 @@ def test_pprs(self):
+                       baseline_timepoint="1",
+                       filter_missing_references=False)
 
         exp_pprs_df = pd.DataFrame({
```
```suggestion
        exp_pprf_df = pd.DataFrame({
```

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -1417,7 +1420,7 @@ def test_pprs(self):
-        pd.testing.assert_frame_equal(sample_pprs_df, exp_pprs_df)
+        pd.testing.assert_frame_equal(pprf_df, exp_pprs_df)
 
     def test_pprs_baseline_sub(self):
```
```suggestion
    def test_pprf_baseline_sub(self):
```

lizgehret commented on q2_fmt/tests/test_engraftment.py:
```diff
@@ -1359,10 +1362,10 @@ def test_peds_no_donor_in_table(self):
+                 reference_column="Ref",
+                 subject_column="subject")
 
     def test_peds_no_donor_in_table_flag(self):
```
```suggestion
    def test_pedf_no_donor_in_table_flag(self):
```

cherman2 committed:
Update q2_fmt/_peds.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/_peds.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/_peds.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/_peds.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/_peds.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/_peds.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/_peds.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/plugin_setup.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/plugin_setup.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/plugin_setup.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/plugin_setup.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/plugin_setup.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/plugin_setup.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/tests/test_engraftment.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Liz's Review

cherman2 committed:
no sample pedf

cherman2 commented:
Thanks for the review. 

I have addressed your suggestions and also I changed the description of detect-donor-indicators to be  detect donor indicator features. Thanks for catching that as well.

lizgehret commented on q2_fmt/_peds.py:
```diff
@@ -316,38 +284,38 @@ def feature_peds(table: pd.DataFrame, metadata: qiime2.Metadata,
-        'title': "Feature PEDS",
+    prdf_df['measure'].attrs.update({
+        'title': "PRDF",
         'description': 'Proportional Engraftment of Donor Strains'
```
```suggestion
        'description': 'Proportional Engraftment of Donor Features'
```

cherman2 replied:
I changed this slightly differently than your suggested because its technically not correct any more for prdf!

lizgehret commented on q2_fmt/_peds.py:
```diff
@@ -365,33 +333,33 @@ def _compute_peds(peds_df: pd.Series, peds_type: str, peds_time: int,
+                               donor_sum[count], time, feature]
+            df = df.dropna()
     else:
         raise KeyError('There was an error finding which PEDS methods to use')
```
```suggestion
        raise KeyError('There was an error finding which proportion method to use')
```

lizgehret commented on q2_fmt/_peds.py:
```diff
@@ -455,51 +423,51 @@ def sample_pprs(table: pd.DataFrame, metadata: qiime2.Metadata,
     })
-    peds_df['measure'].attrs.update({
+    pprf_df['measure'].attrs.update({
         'title': 'PPRS',
```
```suggestion
        'title': 'PPRF',
```

lizgehret reviewed (commented):
ok this looks great @cherman2 - i just found 3 more lil easter eggs, but this is ready to merge after that!

cherman2 committed:
Update q2_fmt/_peds.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
Update q2_fmt/_peds.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

cherman2 committed:
lil diff description
